test/new-e2e/tests/agent-shared-components/api/api_test.go

```diff
@@ -106,10 +314,102 @@ func (v *apiSuite) TestInternalAgentAPIEndpoints() {
 	for _, tc := range testcases {
 		cmd := tc.fetchCommand(authtoken)
 		host := v.Env().RemoteHost
-		require.EventuallyWithT(v.T(), func(t *assert.CollectT) {
-			resp, err := host.Execute(cmd)
-			require.NoError(t, err)
-			assert.Contains(t, resp, tc.want, "%s %s returned: %s, wanted: %s", tc.method, tc.endpoint, resp, tc.want)
-		}, 2*time.Minute, 10*time.Second)
+		v.T().Run(fmt.Sprintf("API - %s test", tc.name), func(t *testing.T) {
+			require.EventuallyWithT(t, func(ct *assert.CollectT) {
+				resp, err := host.Execute(cmd)
+				require.NoError(ct, err)
+				assert.Contains(ct, resp, tc.want, "%s %s returned: %s, wanted match: %s", tc.method, tc.endpoint, resp, tc.want)
+			}, 2*time.Minute, 10*time.Second)
+		})
+	}
+}
+
+func (v *apiSuite) TestSecretsAgentAPIEndpoints() {
+	e := agentEndpointInfo{
+		scheme:   "https",
+		port:     agentCmdPort,
+		endpoint: "/agent/secrets",
+		method:   "GET",
+		data:     "",
+	}
+	want := `used in 'datadog.yaml' configuration in entry 'hostname'`
+
+	config := `secret_backend_command: /tmp/test-secret-api-endpoint/secret-resolver.py
+secret_backend_arguments:
+  - /tmp/test-secret-api-endpoint
+secret_backend_remove_trailing_line_break: true
+secret_backend_command_allow_group_exec_perm: true
+
+log_level: debug
+hostname: ENC[hostname]`
+
+	rootDir := "/tmp/test-secret-api-endpoint"
+	v.Env().RemoteHost.MkdirAll(rootDir)
+	secretResolverPath := filepath.Join(rootDir, "secret-resolver.py")
+
+	v.T().Log("Setting up the secret resolver and the initial api key file")
+
+	secretClient := secrets.NewSecretClient(v.T(), v.Env().RemoteHost, rootDir)
+	secretClient.SetSecret("hostname", "e2e.test")
+
+	v.UpdateEnv(awshost.Provisioner(
+		awshost.WithAgentOptions(
+			secrets.WithUnixSecretSetupScript(secretResolverPath, true),
+			agentparams.WithAgentConfig(config),
+		),
+	))
+
+	authTokenFilePath := "/etc/datadog-agent/auth_token"
+	authtokenContent := v.Env().RemoteHost.MustExecute("sudo cat " + authTokenFilePath)
+	authtoken := strings.TrimSpace(authtokenContent)
+
+	cmd := e.fetchCommand(authtoken)
+
+	require.EventuallyWithT(v.T(), func(ct *assert.CollectT) {
+		resp, err := v.Env().RemoteHost.Execute(cmd)
+		require.NoError(ct, err)
```

Comment:
Same comment here

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/25000/commits/4496c36d67d904500f3638a31c5c24fa0921d50a